### PR TITLE
Preserve module parameters in freezing

### PR DESCRIPTION
**Pull request by bzinodev**
Added preserveParameters to freezing API that allows to preserve module
parameters.

Fixes #{39613}

### CODE REVIEW

File: torch/csrc/jit/passes/freeze_module.cpp [hunk 1]
```diff
@@ -421,6 +424,13 @@ class AttributePropagator {
             if (isEval) {
               attr = overrideGradient(attr);
             }
+            if (preserveParameters_) {
```
Comment (neginraoof) [inline torch/csrc/jit/passes/freeze_module.cpp:427]: We need to enable this for modules in training mode as well.

**Comment by neginraoof** [comment]
Other than training issue, looks good to me. Thanks!

**Review by eellison — COMMENTED** [review]
Looks good to me, one small change to address @neginraoof comment and make this work for training with scriptable models. 

Could we check if the model is in training mode, and if so preserve parameters & betters, and also not inline `getAttr["training"]` nodes ? I think that is all we need to do to make freezing work with training.

File: torch/csrc/jit/passes/freeze_module.cpp (hunk 1 above)
Comment (neginraoof) [reply torch/csrc/jit/passes/freeze_module.cpp]: What's the reason behind keeping the training attribute?

**Review by eellison — APPROVED** [review]
LGTM

**Review by neginraoof — APPROVED** [review]

**Pull request closed without merging**